app/ts/background/iconHandler.ts

```diff
@@ -62,8 +62,13 @@ export async function updateExtensionIcon(websiteTabConnections: WebsiteTabConne
 	return setInterceptorIcon(websiteTabConnections, socket.tabId, ICON_SIGNING, `The Interceptor forwards your transactions to ${ getPrettySignerName(await getSignerName()) } once sent.`)
 }
 
+export function noNewBlockForOverTwoMins(connectionStatus: RpcConnectionStatus) {
+	return connectionStatus && connectionStatus.latestBlock && (connectionStatus.lastConnnectionAttempt.getTime() - connectionStatus.latestBlock.timestamp.getTime()) > 2 * 60 * 1000
```

Comment:
For Ethereum Mainnet, we could be quite a bit more aggressive than this I think.  1 block missing is rare (1% last I heard), 2 blocks in a row missing is even more rare (0.1%), 3 blocks 0.01%, etc.  If this error appears for 12 seconds once every 33 hours that feels fine to me.  We could bump that up 4 blocks in a row (60 seconds) missing to reach once every ~2 weeks (and only for 12 seconds).

Reply:
couple blocks missing can happen by our node lagging a bit behind, I wouldn't want to alert all users if the node lags a bit. I feel this error should happen only when its quite clear the node appears to be frozen